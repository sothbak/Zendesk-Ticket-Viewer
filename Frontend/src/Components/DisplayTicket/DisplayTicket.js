import React, { Component } from 'react';

class DisplayTicket extends Component {
    handleClick = () => {
        this.props.closeDisplayTicket();
    }
    handleClick = this.handleClick.bind(this);

    render() {
        return (
            <div>
                <button onClick={this.handleClick}>Close</button>
                <h1>{this.props.data.subject}</h1>
                <h3>ID : {this.props.data.id}</h3>
                <p>{this.props.data.description}</p>
            </div>
        )
    }
}

export default DisplayTicket;