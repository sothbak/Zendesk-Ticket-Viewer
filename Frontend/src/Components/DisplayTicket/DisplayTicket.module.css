.description {
    text-align: left;
    padding: 0px 50px 50px 50px;
}