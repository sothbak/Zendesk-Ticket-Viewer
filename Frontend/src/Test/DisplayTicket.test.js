import React from 'react';
import { shallow } from 'enzyme';
import DisplayTicket from '../Components/DisplayTicket/DisplayTicket.js';

describe('DisplayTicket', () => {
  it('should render correctly', () => {
    const props = {subject : "subject", id : "id", description : "description", created_at : "2021-05-04T01:07:29Z"};
    const displayTicket = shallow(<DisplayTicket data={props}/>);
    expect(displayTicket).toMatchSnapshot();
  });

  it('should contain the correct creation date', () => {
    const props = {subject : "subject", id : "id", description : "description", created_at : "2021-05-04T01:07:29Z"};
    const displayTicket = shallow(<DisplayTicket data={props}/>);
    expect(displayTicket.text()).toContain("4/5/2021");
  });
});