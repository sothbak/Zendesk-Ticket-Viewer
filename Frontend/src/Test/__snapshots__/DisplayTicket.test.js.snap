// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayTicket should render correctly 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    Close
  </button>
  <h1>
    subject
  </h1>
  <h3>
    ID : 
    id
  </h3>
  <h3>
    Created at : 
    4/5/2021
  </h3>
  <p
    className="description"
  >
    description
  </p>
</div>
`;
